src/restore.ts

```diff
@@ -60,6 +60,6 @@ async function run(): Promise<void> {
     }
 }
 
-run();
+run().then(() => process.exit(0));
```

Comment:
I think this is the wrong approach here. If we're confident that these hanging events are only from error cases, then we should have a corresponding `process.exit` in the `catch` blocks. For save, we'd want a success status code and for restore we'd want a failure status code.

The downside of this current implementation is that we're exiting with a success status code when the step may have failed.

Reply:
Sounds like `process.exit()` is what we want to honor the exit code set by `core.setFailed`
https://github.com/actions/cache/pull/371#issuecomment-658296834